src/sql/src/query_model/hir.rs

```diff
@@ -108,10 +181,14 @@ impl FromHir {
                 BoxScalarExpr::Literal(row.clone(), col_type.clone())
             }
             HirScalarExpr::Column(c) => {
-                assert!(c.level == 0, "correlated columns not yet supported");
+                let context_box = if c.level == 0 {
+                    context_box
+                } else {
+                    self.context_stack[self.context_stack.len() - c.level]
+                };
                 BoxScalarExpr::ColumnReference(self.find_column_within_box(context_box, c.column))
```

Comment:
Add documentation that explains the logic in `find_column_within_box`, as it is not straight-forward to follow.